ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2

```diff
@@ -0,0 +1,119 @@
+{# Parse core_interfaces data model #}
+{% set core_interfaces_data.p2p_interfaces = {} %}
+{% set core_interfaces_data.p2p_port_channels = {} %}
+{% set core_interfaces_data.bgp_neighbors = {} %}
+{% set core_interfaces_data.ospf_interfaces = [] %}
+{% set core_interfaces_data.isis_interfaces = [] %}
+{% for p2p_link in core_interfaces.p2p_links | arista.avd.natural_sort %}
+{%     if p2p_link.profile is arista.avd.defined and core_interfaces.p2p_links_profiles is arista.avd.defined %}
+{%         set p2p_profile = core_interfaces.p2p_links_profiles[p2p_link.profile] | arista.avd.default({}) %}
+{%     else %}
+{%         set p2p_profile = {} %}
+{%     endif %}
+{%     set nodes = p2p_link.nodes | arista.avd.default(
+                       p2p_profile.nodes,
+                       []) %}
+{%     if inventory_hostname in nodes %}
+{# Found this device in core_interfaces. Build interface data #}
+{%         set node_index = nodes.index(inventory_hostname) %}
+{%         set peer_index = 1 - node_index %}
+{%         set ip_pool_name = p2p_link.ip_pool | arista.avd.default(
+                              p2p_profile.ip_pool) %}
+{%         if ip_pool_name is arista.avd.defined and core_interfaces.p2p_links_ip_pools is arista.avd.defined %}
```

Comment:
We can remove the default filter below by doing the check:

```
core_interfaces.p2p_links_ip_pools[ip_pool_name] is arista.avd.defined
```

Reply:
Same concern with `[ ]`. Please test

Reply:
We will let this pass in interest of time.